core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -82,6 +83,18 @@ public ExpireSnapshots expireOlderThan(long timestampMillis) {
     return this;
   }
 
+  @Override
+  public ExpireSnapshots retainLast(int numSnapshots) {
+    Preconditions.checkArgument(1 < numSnapshots,
+            "Number of snapshots to retain must be at least 1, cannot be: %s", numSnapshots);
+    LOG.info("Retaining last {} snapshots and expiring older snapshots", numSnapshots);
+    Snapshot nthSnapshot = findNthSnapshot(numSnapshots);
+    if (nthSnapshot != null) {
+      expireOlderThan(nthSnapshot.timestampMillis());
```

Comment:
This isn't the behavior I would expect from `retainLast`. I would expect to be able to combine this (`retainLast(20)`) with a call to `expireOlderThan(T)`, where snapshots older than T are expired, but at least 20 snapshots are kept. Instead, these two calls would conflict and one would clobber the other.

I would prefer to go with the behavior I described above, where the two configurations work with one another. If one configuration overwrites the other, then we would need to track which one is set and throw an exception if both are called to let the caller know that the API was misused.

I think it's better to avoid a runtime exception by documenting the behavior when the two are used together.

Reply:
I agree, will make the changes to both configurations work with one another and let the caller know if both are called.

Reply:
If both work with one another, there is no need to let the caller know if both are called. I think the right solution is to make the two settings compatible.

Reply:
@rdblue updated to `expireOlderThan` and `retainLast` work with one another. Whichever sets oldest `expireOlderThan` takes precedence.